mlx_graphs/nn/conv/gcn_conv.py

```diff
@@ -55,7 +54,17 @@ def message(
         return x_i if edge_weight is None else edge_weight.reshape(-1, 1) * x_i
 
     def _degree(self, index: mx.array, num_edges: int) -> mx.array:
+        
         out = mx.zeros((num_edges,))
         one = mx.ones((index.shape[0],), dtype=out.dtype)
 
         return mx.scatter_add(out, index, one.reshape(-1, 1), 0)
+    
+    def _deg_inv_sqrt(deg: mx.array) -> mx.array:
+
+        deg += 1e-10
```

Comment:
I'm not sure whether hardcoding this is the best option. It should probably depend on the `dtype` of `deg`?

Reply:
I agree tbh, but this is a workaround suggested by awni as boolean indexing isn't yet available [see](https://github.com/ml-explore/mlx/discussions/408)

Reply:
Yes, what I'm saying is that `1e-10` is not necessarily the best value to use for any `dtype`. For example that's not representable by a float16.

Reply:
What value would you suggest ? 1e-6 is a value that would be representable but actually it's only equal to the expected output rounding to 5 .

Reply:
Yeah, I guess I'd make it depend on `deg.dtype`. What do you think?